content/vgg-classifier-tutorial.md

```diff
@@ -0,0 +1,320 @@
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(MaxPooling2D((2,2), strides=(2,2)))
+
+	model.add(Flatten())
+	model.add(Dense(4096, activation='relu'))
+	model.add(Dropout(0.5))
+	model.add(Dense(4096, activation='relu'))
+	model.add(Dropout(0.5))
+	model.add(Dense(1000, activation='softmax'))
+...
+```
+Then you can load the model:
+```python
+...
+	if weights_path:
+		print("Loading weights:",weights_path)
+		model.load_weights(weights_path)
+
+	return model
+
+mymodel = VGG16(weights_path='vgg16_weights.h5')
+
+```
+
+Modify VGG for your own dataset
+-------
+As you can see on the original VGG16 model train on ImageNet, the output number of classes is ``1000``. But you have to change it for your own dataset. To change it, you just need to change the last ``Dense`` (fully connected) layer of the model to fit with our dataset number of classes (``101`` in this tutorial).
+
+### Change the last layer
+For changing the number of classes of your model, you need to redefine the layer:
+
+```python
+mymodel.layers[-1] = Dense(101, activation='softmax'))
+```
+and reattached it to the before last layer:
+
+```python
+mymodel.layers[-1].set_previous(pretrained_model.layers[-2])
+```
+Nothing more !
+
+### Lock some layers for training
+Because ImageNet is a giant dataset compare to Caltech, the VGG16 pretrained model already know how to recognize the image details, but the model must be adapted/tuned to this dataset. A simple way to do this is to lock  all the convolution layers during the training, meaning the weights of the convolutional layers are never be update. Only the fully connected layers (``Dense``) are updated. But you can easily make some tests by changing the value of: ``unlock_last``. Increasing ``unlock_last`` means: more layers with by updated, decreasing ``unlock_last`` means: less layers are updated.
+
+In keras, it's easy! You just need to iterate on mymodel.layers to set ``trainable`` attribute of the layer to False or True:
+
+```python
+unlock_last = 5
+for num in range(0,len(mymodel.layers)):
+  layer = mymodel.layers[num]
+  if num < len(mymodel.layers)-unlock_last:
+    layer.trainable = False # Locked for the Firsts
+  else:
+    layer.trainable = True # Unlocked for the Lasts
+```
+### Compile the model
+Now, the model can be compile with the categorical cross-entropy loss function  (typically of classification model).
+Note that the learning rate need to be low, else the system is not able to adapted the model for your dataset.
+If it doesn't work for your own dataset, try to decrease first your the learning rate ``lr`` or next increase it (i.e. 0.05) or change the optimizer by `AdaGrad`, `AdaDelta`...
+```python
+sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
+mymodel.compile(optimizer=sgd, loss='categorical_crossentropy')
```

Comment:
In Keras 1.0, we need to pass an argument `metrics=["accuracy"]` to monitor accuracy.